src/lib/engines/layerCompositor/fragment.glsl

```diff
@@ -21,6 +24,26 @@ void main() {
 
   vec4 layerColor = texture2D( layerMap, texelUv, mipmapBias );
 
+  if ( maskMode > 0 ) {
+    
+    vec2 maskTexelUv = ( uvToMaskTexture * vec3( v_uv, 1.0 ) ).xy;
+    vec4 maskTexel = texture2D( maskMap, maskTexelUv, mipmapBias );
+
+    float maskValue;
+
+    if( maskMode == 2 || maskMode == 4 ) {
+      maskValue = maskTexel.r;
```

Comment:
Threekit uses g for opacity but maybe this isn't standard. Could also generalize it by replacing `mask` with `channel`. A quick test on shadertoy tells me that it's actually possible to swizzle via a uniform, like:

uniform int maskChannel;
maskValue[maskChannel]

Reply:
and then have `invert` be another uniform

Reply:
Webgl 1.0 doesn't support non-constant swizzle indices. Nice idea though, will want to use it for when IOS supports 2.0